lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -151,8 +151,37 @@ def which(cmd):
     return None
 
 
-class MPI(object):
+class RunFileLoopMPI(RunFile, RunLoop):
+    """
+    RunFile specific for MPI because intel doesn't feel like fixing their shit
+    """
+    def __init__(self, cmd, **kwargs):
+        self.options = kwargs.pop('options', None)
+        self.counter = 1
+        super(RunFileLoopMPI, self).__init__(cmd, **kwargs)
+        self.readsize = -1
+        self.seen_output = False
+
+    def _loop_process_output(self, output):
+        """
+        check if process is generating any output at all; if not, warn the user after a set amount of time
+        """
+        del output
+        if not self.seen_output:
```

Comment:
I would avoid too deep indentation here, and go with:

```python
# early exit in case we've already seen output
if self.seen_output:
    return

...